roles/network_plugin/contiv/tasks/main.yml

```diff
@@ -0,0 +1,134 @@
+    - {s: "node-{{ inventory_hostname }}-key.pem", d: "proxy-key.pem"}
+
+- name: Contiv | Write contiv-etcd-proxy DaemonSet
+  template:
+    src: contiv-etcd-proxy.yml.j2
+    dest: "{{ kube_config_dir }}/contiv-etcd-proxy.yaml"
+
+- name: Contiv | Write contiv-netplugin DaemonSet
+  template:
+    src: contiv-netplugin.yml.j2
+    dest: "{{ kube_config_dir }}/contiv-netplugin.yaml"
+
+- name: Contiv | Write contiv-netmaster ReplicaSet
+  template:
+    src: contiv-netmaster.yml.j2
+    dest: "{{ kube_config_dir }}/contiv-netmaster.yaml"
+
+- name: Contiv | Write contiv-config ConfigMap
+  template:
+    src: contiv-config.yml.j2
+    dest: "{{ kube_config_dir }}/contiv-config.yaml"
+
+- name: Contiv | Write contiv ClusterRoleBinding
+  template:
+    src: "contiv-{{ item }}-clusterrolebinding.yml.j2"
+    dest: "{{ kube_config_dir }}/contiv-{{ item }}-clusterrolebinding.yaml"
+  with_items:
+    - netmaster
+    - netplugin
+
+- name: Contiv | Write contiv ClusterRole
+  template:
+    src: "contiv-{{ item }}-clusterrole.yml.j2"
+    dest: "{{ kube_config_dir }}/contiv-{{ item }}-clusterrole.yaml"
+  with_items:
+    - netmaster
+    - netplugin
+
+- name: Contiv | Write contiv ServiceAccount
+  template:
+    src: "contiv-{{ item }}-serviceaccount.yml.j2"
+    dest: "{{ kube_config_dir }}/contiv-{{ item }}-serviceaccount.yaml"
+  with_items:
+    - netmaster
+    - netplugin
+
+- name: Contiv | Write contiv-api-proxy DaemonSet
+  template:
+    src: contiv-api-proxy.yml.j2
+    dest: "{{ kube_config_dir }}/contiv-api-proxy.yaml"
+  when: "{{ contiv_enable_api_proxy }}"
+
+- set_fact:
+    contiv_default_encap_mode: "{{ contiv_default_encap_mode }}"
+    contiv_default_gw: "{{ contiv_default_gw }}"
+    contiv_default_network_tag: "{{ contiv_default_network_tag }}"
+    contiv_default_subnet: "{{ contiv_default_subnet }}"
+    contiv_enable_api_proxy: "{{ contiv_enable_api_proxy }}"
+    contiv_fabric_mode: "{{ contiv_fabric_mode }}"
+    contiv_fwd_mode: "{{ contiv_fwd_mode }}"
+    contiv_netmaster_port: "{{ contiv_netmaster_port }}"
+
+- name: Contiv | Create /var/contiv
+  file:
+    path: /var/contiv
+    state: directory
+
+- name: Contiv | Generate contiv-api-proxy certificates
+  script: generate-certificate.sh
+  args:
+    creates: /var/contiv/auth_proxy_key.pem
+  when: "{{ contiv_enable_api_proxy and contiv_generate_certificate }}"
+  delegate_to: "{{ groups['kube-master'][0] }}"
+  run_once: true
+
+- name: Contiv | Fetch the generated certificate
+  fetch:
+    src: "/var/contiv/{{ item }}"
+    dest: "/tmp/kargo-contiv-{{ item }}"
+    flat: yes
+  with_items:
+    - auth_proxy_key.pem
+    - auth_proxy_cert.pem
+  when: "{{ contiv_enable_api_proxy and contiv_generate_certificate }}"
+  delegate_to: "{{ groups['kube-master'][0] }}"
+  run_once: true
+
+- name: Contiv | Copy the generated certificate on nodes
+  copy:
+    src: "/tmp/kargo-contiv-{{ item }}"
+    dest: "/var/contiv/{{ item }}"
```

Comment:
It's the right directory because the configuration of Contiv api proxy use this:

```yaml
            - --tls-key-file=/var/contiv/auth_proxy_key.pem
            - --tls-certificate=/var/contiv/auth_proxy_cert.pem
```
But, I can move the certificates into the `contiv_cert_dir`. I've put the certificate is this directory to follow the kubernetes objects that have already made by the Contiv community.